### PR TITLE
fix: logout should always succeed

### LINKED ISSUE #1863: cli: logout fails if host cert is not trusted
## Describe the bug

``` 
$ dev logout --all --log-level=debug
DEBUG  logging out of all servers

DEBUG  requested but not logged in to server [localhost:8443]
ERROR  POST "/v1/logout": Post "https://localhost/v1/logout": x509: “Infra” certificate is not trusted
```

## Expected behavior

logout should happen whether or not the cert is trusted 

## Environment

```
$ infra version
```

```
$ kubectl version
```

### PR DESCRIPTION
**Pull request by ssoroka**
## Summary

If the /logout api call fails for some reason, we should just delete the access key and continue; we don't really care why it failed, and the key will time out anyway.

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #1863

### CODE REVIEW

File: internal/cmd/logout.go [hunk 1]
```diff
@@ -62,15 +62,15 @@ $ infra logout --all --clear`,
...
 	if !hostConfig.isLoggedIn() {
 		logging.S.Debugf("requested but not logged in to server [%s]", hostConfig.Host)
-		return false, nil
+		return false
 	}
 
 	client, err := apiClient(hostConfig.Host, hostConfig.AccessKey, hostConfig.SkipTLSVerify)
 	if err != nil {
-		return false, err
+		return false
```
Comment (BruceMacD) [inline internal/cmd/logout.go:73]: we should add a debug log for this error just in case

File: internal/cmd/logout.go [hunk 2]
```diff
@@ -80,14 +80,13 @@ func logoutOfServer(hostConfig *ClientHostConfig) (success bool, err error) {
 	err = client.Logout()
 	switch {
 	case api.ErrorStatusCode(err) == http.StatusUnauthorized:
-		logging.S.Warn(err.Error())
-		return false, nil
+		return false
```
Comment (BruceMacD) [inline internal/cmd/logout.go:83]: we should add a debug log for this error too, could be hard to troubleshoot otherwise

Comment (BruceMacD) [inline internal/cmd/logout.go:83]: we should add a debug log for this error too, could be hard to troubleshoot otherwise

**Review by BruceMacD — COMMENTED** [review]
I agree that we should just clear the session on an error response, just some comments around supportability

**Review by jmorganca — APPROVED** [review]

**Review by pdevine — APPROVED** [review]

**Pull request merged**